libvast/src/arrow_table_slice.cpp

```diff
@@ -662,7 +657,9 @@ arrow_table_slice<FlatBuffer>::arrow_table_slice(
       die("failed to deserialize layout: " + render(err));
     state_.layout = type::from_legacy_type(intermediate);
     auto decoder = record_batch_decoder{};
-    state_.record_batch = decoder.decode(slice.schema(), slice.record_batch());
+    state_.record_batch = decoder.decode(
+      as_arrow_buffer(parent->slice(as_bytes(*slice.schema()))),
+      as_arrow_buffer(parent->slice(as_bytes(*slice.record_batch()))));
```

Comment:
TODO: Now that #2020 is merged, rebase on top of recent master and remove the keep alive workaround in the transform engine.